src/DnsUtils.ts

```diff
@@ -0,0 +1,80 @@
+import DnsRecordsError, { DnsRecordsErrorCode } from "./errors/dnsRecordsError";
+import { CryptoRecords, DnsRecord, DnsRecordType } from "./publicTypes";
+import { isStringArray } from "./utils";
+export default class DnsUtils {
+
+  public toList(record: CryptoRecords): DnsRecord[] {
+    const dnsTypes = this.getAllDnsTypes(record);
+    return ([] as DnsRecord[]).concat(
+      ...dnsTypes.map(type => this.constructDnsRecords(record, type))
+    );
+  }
+
+  public toCrypto(records: DnsRecord[]): CryptoRecords {
+    const cryptoRecords:CryptoRecords = {};
+    for (const record of records) {
+      const {type, ttl, value} = record;
+      const ttlInRecord = this.getJsonRepresentation(cryptoRecords[`dns.${type}.ttl`], 'number') as number | undefined;
+      const dnsInRecord = this.getJsonRepresentation(cryptoRecords[`dns.${type}`], 'string[]') as string[] | undefined;
+      if (dnsInRecord) {
+        dnsInRecord.push(value);
+        cryptoRecords[`dns.${type}`] = JSON.stringify(dnsInRecord);
+      } else {
+        cryptoRecords[`dns.${type}`] = JSON.stringify([value]);
+        cryptoRecords[`dns.${type}.ttl`] = ttl.toString(10);
+      }
+
+      if (!!ttlInRecord && ttlInRecord !== ttl) {
+        throw new DnsRecordsError(DnsRecordsErrorCode.InconsistentTtl, {recordType: type});
+      }
+    }
+    return cryptoRecords;
+  }
+
+  private getJsonRepresentation(rawRecord: string | undefined, type:  'string[]' | 'number'): string[] | number | undefined {
```

Comment:
I don't see much sense into making one method with `type` argument instead of 2 methods.